File: py/src/rulecat/rule_meta.py
from __future__ import annotations
from dataclasses import dataclass, field
from typing import Any


@dataclass
class RuleMeta:
    name: str
    qualifier: float
    strength: int  # Strength enum value
    defeats: list[str] = field(default_factory=list)
    specs: list[Any] = field(default_factory=list)
    fn: Any = None  # RuleFunc
    on_active: Any = None     # NodeHandler | None
    on_defeated: Any = None
    on_inactive: Any = None
    run_graph: Any = None     # "Graph | None" — Active면 Run할 하위 그래프 (순환 import 회피로 Any)

File: ts/src/rule-meta.ts
import type { Strength, Specs, RuleFunc, NodeHandler } from "./types.js";
import type { Graph } from "./graph.js";   // 타입 전용(순환 import 안전)
export interface RuleMeta {
  name: string;
  qualifier: number;
  strength: Strength;
  defeats: string[];
  specs: Specs;
  fn: RuleFunc;
  runOn?: NodeHandler;      // Active(activated && verdict>0) 시 발화하는 핸들러
  runGraph?: Graph;         // 이 노드가 Active면 Run할 하위 그래프
}
